### ISSUE TITLE
UWP compatibility

### ISSUE THREAD

**Issue by bc3tech** [googleads/googleads-dotnet-lib#51]
[This package](https://www.nuget.org/packages/Google.Dfp/18.15.0) isn't compatible with Windows 10 Universal projects at this time. I've a need for this, so I'm filing an issue.

**Comment by Englbach** [comment]
Me too.

**Comment by AnashOommen** [comment]
Hi,
I haven't tried this yet, but I'll accept this as an item for further investigation.

**Comment by Sonictrio** [comment]
I have a need for support for UWP app too.

**Comment by jimper** [comment]
I've spent some time looking into this and unfortunately it doesn't seem feasible for us to make a UWP compatible version of this package. Our library has dependencies on a number of types (namely System.Web.*), which are not supported by UWP.

**Comment by bc3tech** [comment]
that doesn't seem quite right given (a google rep's affirmation that it can be done w/ straight HTTP calls)[https://lh3.googleusercontent.com/-yeOL6sATi0E/ViULxaU9DtI/AAAAAAAACqc/_xc3u3FsAxc/s1600/2015-10-19_1025.png].

**Comment by jimper** [comment]
That may be true, and in fact there's been a lot of work done on HttpClient in .NET Core 5. However, that does not change the fact that this library is dependant on the System.Web namespace (among others), which is unsupported by .NET for UWP apps.

You can find a list of the UWP supported namespaces in the [MSDN docs](https://msdn.microsoft.com/en-us/library/mt185501.aspx).

**Comment by Narwalter** [comment]
Opened issue #56 to migrate to WCF. That's a big blocker for UWP support.

**Comment by bc3tech** [comment]
thanks @christopherseeley. If WCF migration is a stepping stone to UWP, then should this issue really be closed vs "postponed" or "tabled" or "dependent upon #56 "?

**Comment by AnashOommen** [comment]
It's not just WCF, that just addresses about 30-40% of the problem. And WCF has significantly different stub classes than System.We

 Then there's System.Web (for which we could add HTTPClient replacement)

**Comment by bc3tech** [comment]
Regarding 2, 4, and 5:
* UWP does still have a very DOM-centric XML object: `Windows.Data.Xml.Dom.XmlDocument`
* Might `HttpWebRequest` be able to be migrated to `HttpRequestMessage`?
* Utilizing [BouncyCastle](https://www.nuget.org/packages/Portable.BouncyCastle/) you can still do cryptography.

With 200 million Windows 10 installs & counting, I'd highly recommend raising this from low priority.

**Comment by patricktedeschieldorado** [comment]
Thinking of basic use of DFP to display banners on UWP apps, is there any way to achieve that by exchanging some SOAP messages "manually", without using any library?

**Comment by bc3tech** [comment]
@patricktedeschieldorado i'm told by other parties that some UWP devs have implemented DFP using a WebView and the REST API, but I've yet to see that made in to a reusable component (unfortunately). I may start down this path soon, though, if nothing official is announced.

**Comment by AnashOommen** [comment]
Patrick,

To make a call manually, you need to post the SOAP body manually to the DFP endpoint as shown in this wiki: https://github.com/googleads/googleads-dotnet-lib/wiki/NoClientLibrary#dfp-api-get-all-users

It is doable, but not very scalable unless you don't mind building the XML from scratch every time.

This item is on the top of our todo list, and have some work in progress. Unfortunately, the requirements of UWP is pretty tough to meet with all the libraries we can find around. So this is taking longer than we'd like to.

**Comment by Sonictrio** [comment]
Hi there. 

To get a DfP advert implementation in our UWP app, we took the [Simplified URL tags](https://support.google.com/dfp_premium/answer/2623168?hl=en) route. This generates a link and requests an image.  We then have a custom control that is dropped in where we need adverts, and can just pass in the relevant information that position would need when building the advert url.

We looked at number of solutions, but non gave us the confidence to move forward with. Although this isn't ideal, it enabled us to get simple adverts into the app without more hold up.

**Comment by christopherseeley** [comment]
@Sonictrio this library is not meant for serving DFP ads either in apps or on websites. It's for the DFP [SOAP API](https://developers.google.com/doubleclick-publishers/), which deals with sales management, inventory management, reporting etc.

Are most of you looking to display DFP ads in UWP apps?

**Comment by Sonictrio** [comment]
*nods and quietly walks away* :-(

**Comment by bc3tech** [comment]
@christopherseeley I can only speak for myself and say "yes" - the .Net lib covered by this repo **is** the one that is used to display ads in other .Net platforms, is it not??

**Comment by christopherseeley** [comment]
No, this lib does not deal with displaying ads from DFP. Unfortunately I'm unaware of a dedicated SDK for requesting DFP ads on .Net platforms, UWP or otherwise. The Simplified URL tags mentioned by @Sonictrio are the best option at this time.

**Comment by SonicGD** [comment]
Hello everyone! Any news on this issue? We are really want to manage AdWords from our .NET Core application on Linux. 

For now we are running it on mono and it seems to be working, but you never know =)

**Comment by SonicGD** [comment]
Hello everyone! Any news on this issue? We are really want to manage AdWords from our .NET Core application on Linux. 

For now we are running it on mono and it seems to be working, but you never know =)

**Comment by christopherseeley** [comment]
We're still working on upgrading to WCF and removing other blockers, but we haven't forgotten about this.

**Comment by christopherseeley** [comment]
Marking this as a duplicate of #96. The lib will get UWP 10.0.16299 compatibility via .NET Standard 2.0
